src/rust/ide/view/src/project.rs

```diff
@@ -90,26 +128,75 @@ pub struct View {
 impl View {
     /// Constructor.
     pub fn new(app:&Application) -> Self {
-        let model = Model::new(app);
-        let frp   = Frp::new_network();
 
-        let network = &frp.network;
+        let model             = Model::new(app);
+        let frp               = Frp::new_network();
+        let searcher          = &model.searcher.frp;
+        let graph             = &model.graph_editor.frp;
+        let network           = &frp.network;
+        let searcher_left_top = Animation::<Vector2<f32>>::new(network);
 
         frp::extend!{ network
+
             // === Documentation Set ===
 
             eval frp.set_documentation_data ((data) model.documentation.frp.send_data.emit(data));
+            eval frp.set_suggestions        ((provider) model.searcher.frp.set_entries(provider));
+
 
+            // === Searcher Position ===
 
-            // === Documentation toggle ===
+            _eval <- all_with(&searcher_left_top.value,&searcher.size,f!([model](lt,size) {
+                let x = lt.x + size.x / 2.0;
+                let y = lt.y - size.y / 2.0;
+                model.searcher.set_position_xy(Vector2(x,y));
+            }));
 
-            let documentation_press_ev = frp.press_documentation_view_visibility.clone_ref();
-            let documentation_release  = frp.release_documentation_view_visibility.clone_ref();
-            documentation_pressed            <- bool(&documentation_release,&documentation_press_ev);
-            documentation_was_pressed        <- documentation_pressed.previous();
-            documentation_press              <- documentation_press_ev.gate_not(&documentation_was_pressed);
-            documentation_press_on_off       <- documentation_press.map(f_!(model.is_documentation_visible()));
-            frp.source.documentation_visible <+ documentation_press_on_off;
+
+            // === Editing ===
+
+            // The order of instructions below is important to properly distinguish between
+            // committing and aborting node editing.
+
+            // This node is false when received "abort_node_editing" signal, and should get true
+            // once processing of "edited_node" event from graph is performed.
+
+            editing_aborted <- any(...);
+            editing_aborted <+ frp.abort_node_editing.constant(true);
+            should_finish_editing <- any(frp.abort_node_editing,searcher.editing_committed
+                ,frp.add_new_node);
```

Comment:
could you please break the line after `<-` instead?

Reply:
no problem